src/openrct2/CmdlineSprite.cpp

```diff
@@ -114,55 +117,33 @@ std::optional<SpriteFile> SpriteFile::Open(const utf8* path)
 
 bool SpriteFile::Save(const utf8* path)
 {
-    FILE* file = fopen(path, "wb");
-    if (file == nullptr)
-        return false;
-
-    if (fwrite(&Header, sizeof(rct_sprite_file_header), 1, file) != 1)
-    {
-        fclose(file);
-        return false;
-    }
-
-    if (Header.num_entries > 0)
+    try
     {
-        std::unique_ptr<rct_g1_element_32bit[]> saveElements = std::make_unique<rct_g1_element_32bit[]>(Header.num_entries);
-        if (saveElements == nullptr)
-        {
-            fclose(file);
-            return false;
-        }
+        OpenRCT2::FileStream stream(path, OpenRCT2::FILE_MODE_WRITE);
+        stream.Write(&Header, sizeof(rct_sprite_file_header));
 
-        for (uint32_t i = 0; i < Header.num_entries; i++)
+        if (Header.num_entries > 0)
         {
-            rct_g1_element* inElement = &Entries[i];
-            rct_g1_element_32bit* outElement = &saveElements[i];
-
-            outElement->offset = static_cast<uint32_t>(
-                (reinterpret_cast<uintptr_t>(inElement->offset) - reinterpret_cast<uintptr_t>(Data.data())));
-            outElement->width = inElement->width;
-            outElement->height = inElement->height;
-            outElement->x_offset = inElement->x_offset;
-            outElement->y_offset = inElement->y_offset;
-            outElement->flags = inElement->flags;
-            outElement->zoomed_offset = inElement->zoomed_offset;
-        }
+            bool wasAbsolute = isAbsolute;
+            if (wasAbsolute)
+            {
+                MakeEntriesRelative();
+            }
 
-        if (fwrite(saveElements.get(), Header.num_entries * sizeof(rct_g1_element_32bit), 1, file) != 1)
-        {
-            fclose(file);
-            return false;
-        }
+            stream.Write(Entries.data(), sizeof(rct_g1_element) * Header.num_entries);
+            stream.Write(Data.data(), Header.total_size);
 
-        if (fwrite(Data.data(), Header.total_size, 1, file) != 1)
-        {
-            fclose(file);
-            return false;
+            if (wasAbsolute)
+            {
+                MakeEntriesAbsolute();
+            }
```

Comment:
(On commit https://github.com/OpenRCT2/OpenRCT2/pull/13590/commits/48b3fe8944c9ece0036c71c037941a7fbc2ada17)

If an exception is thrown, we never make it absolute again. I'd use an RAII solution for this
```c++
ScopedRelativeSpriteFile(SpriteFile& sFile, bool isAbsolute)
    : Spritefile(sFile), WasAbsolute(isAbsolute)
{
    if (WasAbsolute)
        sFile.MakeEntriesRelative();
}

~ScopedRelativeSpriteFile()
{
    if (WasAbsolute)
        sFile.MakeEntriesAbsolute();
}
```

And then just instantiate this before writing to the stream.